challenge_demo/scripts/demo.py

```diff
@@ -132,14 +132,13 @@ def main():
                     else:
                         sentence, semantics = robot.picovoice.get_intent("demo", demo=True)
                         semantics = create_semantics(semantics)
-                    timeout_count = 0
                     break
                 except hmi.TimeoutException:
                     robot.speech.speak(random.sample(knowledge.not_understood_sentences, 1)[0])
-                    timeout_count += 1
-                    if timeout_count >= 3:
-                        rospy.logwarn("[GPSR] Timeout_count: {}".format(timeout_count))
-                        break
+            else:
+                rospy.logwarn("[GPSR] speech timed out {timeout_tries} times, aborting")
+                robot.speech.speak("I am sorry but I cannot hear you. Let's not try again.")
+                continue  # Continue the outer while loop
 
             # check if we have heard this correctly
             robot.speech.speak('I heard %s, is this correct?' % sentence)
```

Comment:
Something I noticed but did not fix. The robot will say the action it understood rather than what was said. i.e. it will say "I heard hand-over is this correct?" or "I heard navigate-to, is this correct? or "I heard say, is this correct"

Reply:
Yes, I think this is the new behaviour. But @lottemessing should confirm this.